tests/system/Router/RouterTest.php

```diff
@@ -575,4 +723,35 @@ public function testRegularExpressionPlaceholderWithUnicode()
 		];
 		$this->assertEquals($expected, $router->params());
 	}
+
+	public function testRouterPriorDirectory()
+	{
+		$router = new Router($this->collection, $this->request);
+
+		$router->setDirectory('foo/bar/baz', false, true);
+		$router->handle('Some_controller/some_method/param1/param2/param3');
+
+		$this->assertEquals('foo/bar/baz/', $router->directory());
```

Comment:
@sneakyimp Why is this test necessary?
Why does `setDirectory()` take precedence over URI?

Reply:
I've found your answer.
https://github.com/codeigniter4/CodeIgniter4/pull/4307/files#r584375184